pandas/tests/test_strings.py

```diff
@@ -642,6 +642,129 @@ def test_extract_single_series_name_is_preserved(self):
         tm.assert_series_equal(r, e)
         self.assertEqual(r.name, e.name)
 
+    def test_extractall(self):
+        subject_list = [
+            'dave@google.com',
+            'tdhock5@gmail.com',
+            'maudelaperriere@gmail.com',
+            'rob@gmail.com some text steve@gmail.com',
+            'a@b.com some text c@d.com and e@f.com',
+            np.nan,
+            "",
+        ]
+        expected_tuples = [
+            ("dave", "google", "com"),
+            ("tdhock5", "gmail", "com"),
+            ("maudelaperriere", "gmail", "com"),
+            ("rob", "gmail", "com"), ("steve", "gmail", "com"),
+            ("a", "b", "com"), ("c", "d", "com"), ("e", "f", "com"),
+        ]
+        named_pattern = r'''
+        (?P<user>[a-z0-9]+)
+        @
+        (?P<domain>[a-z]+)
+        \.
+        (?P<tld>[a-z]{2,4})
+        '''
+        expected_columns = ["user", "domain", "tld"]
+        S = Series(subject_list)
+        # extractall should return a DataFrame with one row for each
+        # match, indexed by the subject from which the match came.
+        expected_index = MultiIndex.from_tuples([
+            (0, 0),
+            (1, 0),
+            (2, 0),
+            (3, 0),
+            (3, 1),
+            (4, 0),
+            (4, 1),
+            (4, 2),
+        ], names=(None, "match"))
+        expected_df = DataFrame(
+            expected_tuples, expected_index, expected_columns)
+        computed_df = S.str.extractall(named_pattern, re.VERBOSE)
+        tm.assert_frame_equal(computed_df, expected_df)
+        # The index of the input Series should be used to construct
```

Comment:
put blank lines between different tests here (or sections of tests), makes more readable